src/presence_analyzer/tests.py

```diff
@@ -48,11 +60,60 @@ def test_api_users(self):
         """
         resp = self.client.get('/api/v1/users')
         self.assertEqual(resp.status_code, 200)
+
         self.assertEqual(resp.content_type, 'application/json')
         data = json.loads(resp.data)
         self.assertEqual(len(data), 2)
         self.assertDictEqual(data[0], {u'user_id': 10, u'name': u'User 10'})
 
+    def test_presence_weekday_non_existent_id(self):
+        """
+        Test data contains ids: 10 and 11. Giving number 1 should result in 404 exit code.
+        """
+        response = self.client.get('/api/v1/presence_weekday/1')
+        self.assertEqual(response.status_code, 404)
+
+    def test_presence_weekday_view(self):
+        """
+        Test json response for user 10.
+        """
+        response = self.client.get('/api/v1/presence_weekday/10')
+
+        data = json.loads(response.data)
+        weekdays = utils.group_by_weekday(utils.get_data()[10])
+        expected_data = [
+            [calendar.day_abbr[weekday], sum(intervals)]
+            for weekday, intervals in enumerate(weekdays)
+            ]
+        expected_data.insert(0, ['Weekday', 'Presence (s)'])
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(expected_data, data)
+
+    def test_mean_time_non_existent_id(self):
+        """
+        Test data contains ids: 10 and 11. Giving number 1 should result in 404 exit code.
+        """
+        response = self.client.get('/api/v1/mean_time_weekday/1')
+
+        self.assertEqual(response.status_code, 404)
+
+    def test_mean_time_weekday_view(self):
+        """
+        Test json response for user 10.
+        """
+        response = self.client.get('/api/v1/mean_time_weekday/10')
+
+        data = json.loads(response.data)
+        weekdays = utils.group_by_weekday(utils.get_data()[10])
+        expected_data = [
+            [calendar.day_abbr[weekday], utils.mean(intervals)]
+            for weekday, intervals in enumerate(weekdays)
+            ]
```

Comment:
expected_data = [
        [calendar.day_abbr[weekday], utils.mean(intervals)]
        for weekday, intervals in enumerate(weekdays)
    ]